src/Tests/dotnet-workload-search.Tests/GivenDotnetWorkloadSearch.cs

```diff
@@ -0,0 +1,117 @@
+            {
+                new WorkloadDefinition(new WorkloadDefinitionId("mock-workload-1"), false, null, WorkloadDefinitionKind.Dev, new List<WorkloadDefinitionId>(), 
+                    new List<WorkloadPackId>(), new List<string>()),
+                new WorkloadDefinition(new WorkloadDefinitionId("mock-workload-2"), false, string.Empty, WorkloadDefinitionKind.Build, null, 
+                    new List<WorkloadPackId>(), new List<string>() { "platform1", "platform2" }),
+                new WorkloadDefinition(new WorkloadDefinitionId("mock-workload-3"), true, "Fake description 1", WorkloadDefinitionKind.Dev, 
+                    new List<WorkloadDefinitionId>() { new WorkloadDefinitionId("mock-workload-2") }, new List<WorkloadPackId>(), new List<string>()),
+                new WorkloadDefinition(new WorkloadDefinitionId("fake-workload-1"), true, null, WorkloadDefinitionKind.Build, 
+                    new List<WorkloadDefinitionId>(), new List<WorkloadPackId>(), null),
+                new WorkloadDefinition(new WorkloadDefinitionId("fake-workload-2"), false, "Fake description 2", WorkloadDefinitionKind.Dev, null, 
+                    new List<WorkloadPackId>(), new List<string>())
+            };
+
+        public GivenDotnetWorkloadSearch(ITestOutputHelper log) : base(log)
+        {
+            _reporter = new BufferedReporter();
+        }
+
+        [Fact]
+        public void GivenNoWorkloadsAreInstalledSearchIsEmpty()
+        {
+            _reporter.Clear();
+            var parseResult = Parser.GetWorkloadsInstance.Parse("dotnet workload search");
+            var workloadResolver = new MockWorkloadResolver(new List<WorkloadDefinition>());
+            var command = new WorkloadSearchCommand(parseResult, _reporter, workloadResolver, "6.0.100");
+            command.Execute();
+
+            _reporter.Lines.Count.Should().Be(4, because: "Output should have header and no values.");
+        }
+
+        [Fact]
+        public void GivenNoStubIsProvidedSearchShowsAllWorkloads()
+        {
+            _reporter.Clear();
+            var parseResult = Parser.GetWorkloadsInstance.Parse("dotnet workload search");
+            var workloadResolver = new MockWorkloadResolver(_avaliableWorkloads);
+            var command = new WorkloadSearchCommand(parseResult, _reporter, workloadResolver, "6.0.100");
+            command.Execute();
+
+            var output = string.Join(" ", _reporter.Lines);
+            foreach (var workload in _avaliableWorkloads)
+            {
+                output.Contains(workload.Id.ToString()).Should().BeTrue();
+                if (workload.Description != null)
+                {
+                    output.Contains(workload.Description).Should().BeTrue();
+                }
+                output.Contains(workload.Kind.ToString()).Should().BeFalse();
+            }
+        }
+
+        [Fact]
+        public void GivenDetailedVerbositySearchShowsAllColumns()
```

Comment:
We should show the output sample to Kathleen in case we need more info to display like the version.

Reply:
Discussed with Kathleen and we're good to go for now. We'll bring up wording for the column names at a future meeting.